lib/jxl/blending.cc

```diff
@@ -10,6 +10,60 @@
 
 namespace jxl {
 
+namespace {
+
+// Given two rects A and B, returns a set of rects whose union is A \ B. This
+// may require from 0 to 4 rects, one for each non-empty side of B. The order is
+// consistent when called successively with "parallel" rects.
+//
+// +----------------------+
+// |         top          |
+// +------+-------+-------+
+// | left | inner | right |
+// +------+-------+-------+
+// |        bottom        |
+// +----------------------+
+std::vector<Rect> SubtractRect(const Rect& outer, const Rect& inner) {
```

Comment:
Maybe I'm just over-optimizing, but how about filling an output array and returning the number?

Reply:
I was thinking of that too but wasn’t sure whether it would be worth it. Let me give it a try.

Reply:
There we go.

Reply:
(And just removed the leftover `std::vector<Rect> result;`.)